py/load_firmware.py

```diff
@@ -48,43 +66,58 @@ def main() -> int:
         eprint("Expecting signed firmware")
         return 1
 
-    bootloaders = devices.get_bitbox02_devices(devices.BB02_BOOTLOADER)
-    bootloaders.extend(devices.get_bitbox02_devices(devices.BB02BTC_BOOTLOADER))
-    bitboxes = devices.get_bitbox02_devices()
-
-    def _wait_for_bootloaders() -> List[devices.DeviceInfo]:
-        while True:
-            bootloaders = devices.get_bitbox02_devices(devices.BB02_BOOTLOADER)
-            bootloaders.extend(devices.get_bitbox02_devices(devices.BB02BTC_BOOTLOADER))
-            if bootloaders:
-                return bootloaders
-            sys.stdout.write(".")
-            sleep(1)
-
-    if not bootloaders:
-        if len(bitboxes) != 1:
-            eprint(
-                "No bitbox02 bootloader detected. Insert exactly one bootloader or "
-                "bitbox02 device."
-            )
-            return 1
+    bootloader = None
+    try:
+        bootloader = devices.get_bitbox02_bootloader()
+    except TooManyFoundException:
+        eprint("Found multiple bb02 bootloader standard editions. Only one supported.")
+
+    if bootloader is None:
+        try:
+            bootloader = devices.get_bitbox02btc_bootloader()
+        except TooManyFoundException:
+            eprint("Found multiple bb02 bootloader btc editions. Only one supported.")
+
+    if bootloader is None:
+        bitbox = None
+        try:
+            bitbox = devices.get_bitbox02_device()
+        except TooManyFoundException:
+            eprint("Found multiple bb02 standard editions. Only one supported.")
+
+        if bitbox is None:
+            try:
+                bitbox = devices.get_bitbox02btc_device()
+            except TooManyFoundException:
+                eprint("Found multiple bb02 btc editions. Only one supported.")
+
+        if bitbox is None:
+            eprint("Neither bootloader nor bitbox found.")
 
         # bitbox02 detected -> send command to reboot into bootloader to upgrade.
-        def show_pairing(code: str) -> None:
-            eprint("Please compare and confirm the pairing code on your BitBox02:")
-            eprint(code)
+        def _show_pairing(code: str) -> None:
+            print("Please compare and confirm the pairing code on your BitBox02:")
+            print(code)
 
-        bitbox = BitBox02(device_info=bitboxes[0], show_pairing_callback=show_pairing)
+        bitbox = BitBox02(device_info=bitbox, show_pairing_callback=_show_pairing)
         bitbox.reboot()
-        bootloaders = _wait_for_bootloaders()
 
-    if len(bootloaders) > 1:
-        eprint("Multiple bootloaders detected. Only one supported")
-        return 1
+        # wait for it to reboot
+        while True:
```

Comment:
if you have multiple it might take the wrong one

Reply:
ack